### PR TITLE
chore(stream): add StackSetStreamer

### PR DESCRIPTION
**Pull request by efekarakus**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/aws/cloudformation/stackset/status.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+func (s OpStatus) IsSuccessful() bool {
+	return s == opStatusSucceeded
+}
+
+// IsFailure returns true if the operation terminated in failure.
+func (s OpStatus) IsFailure() bool {
+	return s == opStatusStopped || s == opStatusFailed
+}
+
+// InstanceStatus represents a stack set's instance detailed status.
```
Comment (efekarakus) [inline internal/pkg/aws/cloudformation/stackset/status.go:41]: This file is the same as `opstatus.go` that got deleted, I just renamed the file and added the `InstanceStatus` type

File: internal/pkg/aws/cloudformation/stackset/status_test.go [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+package stackset
+
+import (
+	"testing"
+
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/stretchr/testify/require"
+)
+
+func TestOpStatus_IsCompleted(t *testing.T) {
```
Comment (efekarakus) [inline internal/pkg/aws/cloudformation/stackset/status_test.go:13]: same here, all the `TestOpStatus_` tests are the same as before, I added the `TestInstanceStatus_` tests

File: internal/pkg/aws/cloudformation/stackset/stackset.go [hunk 3]
```diff
@@ -191,28 +191,45 @@ func (ss *StackSet) CreateInstancesAndWait(name string, accounts, regions []stri
...
-			Region:  aws.StringValue(summary.Region),
-		})
+	for {
+		in := &cloudformation.ListStackInstancesInput{
+			StackSetName: aws.String(name),
+			NextToken:    nextToken,
+		}
+		for _, opt := range opts {
+			opt(in)
+		}
```
Comment (dannyrandall) [inline internal/pkg/aws/cloudformation/stackset/stackset.go]: could just do this once outside of the for loop and then just update `in.NextToken`

File: internal/pkg/stream/cfnstackset.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	lastSentOp stackset.Operation
+	curOp      stackset.Operation
+
+	clock   clock
+	rand    func(int) int
+	retries int
+}
+
+// NewStackSetStreamer creates a StackSetStreamer for the given stack set name and operation.
+func NewStackSetStreamer(cfn StackSetDescriber, ssName, opID string, opStartTime time.Time) *StackSetStreamer {
```
Comment (dannyrandall) [inline internal/pkg/stream/cfnstackset.go:49]: `ssName` does this function take my identity

File: internal/pkg/stream/cfnstackset.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+	DescribeOperation(name, opID string) (stackset.Operation, error)
+}
+
+// StackSetOpEvent represents a stack set operation status update message.
+type StackSetOpEvent struct {
+	Name      string // The name of the stack set.
+	Operation stackset.Operation
+}
+
+// StackSetStreamer is a [Streamer] emitting [StackSetOpEvent] messages for instances under modification.
```
Comment (dannyrandall) [inline internal/pkg/stream/cfnstackset.go:29]: oooh exciting doc features

File: internal/pkg/stream/cfnstackset.go [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+type StackSetStreamer struct {
+	stackset    StackSetDescriber
+	ssName      string
+	opID        string
+	opStartTime time.Time
+
+	mu         sync.Mutex
```
Comment (dannyrandall) [inline internal/pkg/stream/cfnstackset.go]: is this only for subs? if so, can we call it 
```suggestion
	subsMu         sync.Mutex
```
or just add a doc string saying it protects `subs`?
?

File: internal/pkg/aws/cloudformation/stackset/stackset.go (hunk 3 above)
Comment (efekarakus) [reply internal/pkg/aws/cloudformation/stackset/stackset.go]: Done!

File: internal/pkg/stream/cfnstackset.go (hunk 6 above)
Comment (efekarakus) [reply internal/pkg/stream/cfnstackset.go]: Done!

**Review by iamhopaul123 — APPROVED** [review]
🚢

**Review by dannyrandall — APPROVED** [review]

**Pull request merged**